### ISSUE TITLE
PHPStan issue

### ISSUE THREAD

**Issue by rjensen26** [forumone/gesso#806]
When setting up a new theme there are some common PHPStan things that come up and should be fixed.

<img width="1259" alt="Screenshot 2024-03-06 at 9 30 23 AM" src="https://github.com/forumone/gesso/assets/5650670/aa399a7b-55f1-430d-a36a-b803e81b6b45">

**Comment by timnolte** [comment]
FYI, just a note that PHPStan has updates regularly that fix issues that weren't being flagged before. This just means that this sort of thing is not a 1-and-done thing but something that requires on-going maintenance. Folks should be opening a PR with any needed updates. There is an aspect where linting and code quality checks should be added to the theme repos, which isn't in place, so that really should be the first thing done.

**Comment by rjensen26** [comment]
Here is how to resolve it: https://github.com/forumone/wvde/commit/9c6aaf0c715788f4e6796e71c2f35d9811ae9030

**Comment by dcmouyard** [comment]
I’m not a fan of how PHPStan differs from the Drupal coding standards. Running it on core Drupal files produces tons of errors, whereas running phpcs does not.

> There is an aspect where linting and code quality checks should be added to the theme repos, which isn't in place, so that really should be the first thing done.

Now that there are commands for `phpstan` and `phpcs` in the latest [Drupal project template](https://github.com/forumone/drupal-project) I can easily run these tests when working on Gesso, but I’d love for there to be something that automatically runs whenever a PR is created.

**Comment by timnolte** [comment]
@dcmouyard so this is odd as I thought that Drupal Core used PHPStan already, and there is a Drupal specific PHPStan package that is being installed in Drupal projects. (https://github.com/forumone/drupal-project/blob/main/composer.json#L72)

I'm curious why you would be scanning Drupal core files with PHPStan on any of our projects?

I did just look and Drupal Core uses the same PHPStan Drupal package we are using. https://git.drupalcode.org/project/drupal/-/blob/11.x/composer.json?ref_type=heads#L26

There are many ways to configured PHPStan. On if the goals here by Ryan is to get linting in place so that we can get them to run in PRs as well just like on projects.

**Comment by dcmouyard** [comment]
@timnolte 

> I'm curious why you would be scanning Drupal core files with PHPStan on any of our projects?

I don’t think we should be scanning core Drupal files and I’m not aware of any projects doing this. While testing this PR, though, I copied the `olivero.theme` file from the Drupal core theme into Gesso to see if `ddev phpstan` would output any errors, and it found 79.

**Comment by dcmouyard** [comment]
Perhaps we should have PHPStan ignore the `includes/` directory?

**Comment by timnolte** [comment]
@dcmouyard I'm not clear I understand why the `includes` directory would be excluded?

**Comment by dcmouyard** [comment]
@timnolte Because that’s where we have our theme functions and preprocess functions in Gesso, and those don’t seem to use the same PHPStan standards that Drupal follows elsewhere. Everything in `gesso_helper` definitely should, since that is more _module_ rather than _theme_ code.

**Comment by timnolte** [comment]
@dcmouyard I think this is more than likely a configuration issue. All custom code should be checked against both the PHPCS & PHPStan Drupal standards. This repo doesn't even have either a specific PHPCS or PHPStan configuration setup like the Drupal projects has:
- https://github.com/forumone/drupal-project/blob/main/.phpcs.xml.dist
- https://github.com/forumone/drupal-project/blob/main/phpstan.neon.dist

Actual configuration should also be added to this repo so that things are all checked correctly.

**Comment by timnolte** [comment]
@rjensen26 are you going to actually add in PHPCS & PHPStan linting setup into Gesso? I think this issue and the linked PR are just code fixes but not actually putting in the checking to keep future theme repo work updated, correct?

**Comment by dcmouyard** [comment]
@rjensen26 @timnolte I think I figured out the issue. The `phpstan.neon.dist` file in Drupal Project doesn’t match what’s in Drupal. I’ll create a PR over there to get this change in place.

**Comment by dcmouyard** [comment]
PR is here: forumone/drupal-project#113
